src/Tes.Runner/Docker/DockerExecutor.cs

```diff
@@ -193,22 +194,31 @@ private async Task<MultiplexedStream> StreamStdOutAndErrorAsync(string container
         }
 
         private async Task<CreateContainerResponse> CreateContainerAsync(string imageWithTag,
-            List<string> commandsToExecute, List<string>? volumeBindings, string? workingDir)
+            List<string> commandsToExecute, List<string>? volumeBindings, string? workingDir, bool? gpus = default)
         {
             logger.LogInformation(@"Creating container with image name: {ImageWithTag}", imageWithTag);
 
             var createResponse = await dockerClient.Containers.CreateContainerAsync(
-                new CreateContainerParameters
+                new()
                 {
                     Image = imageWithTag,
                     Cmd = commandsToExecute,
                     AttachStdout = true,
                     AttachStderr = true,
                     WorkingDir = workingDir,
-                    HostConfig = new HostConfig
+                    HostConfig = new()
                     {
                         AutoRemove = true,
-                        Binds = volumeBindings
+                        Binds = volumeBindings,
+                        DeviceRequests = gpus.GetValueOrDefault()
+                            ? [new()
+                            {
+                                Driver = "nvidia",
+                                Count = -1,
```

Comment:
should we default to 1?

Reply:
-1 means "all". There's no reason to arbitrarily limit access from GPUs the customer will be paying for.